src/assignment.js

```diff
@@ -158,38 +166,45 @@ function createAssignmentField(checkFor, item, fieldText, defaultState) {
   return assignmentsToggleField;
 }
 
-function getInitialState(item, defaultState) {
-  const itemState = localStorage.getItem(item);
-
+function getInitialState(item, defaultState, checkFor) {
+  const itemState = JSON.parse(localStorage.getItem(item))[1];
   if (itemState === "true") {
     return true;
   } else if (itemState === "false") {
     return false;
   }
 
-  localStorage.setItem(item, defaultState);
+  localStorage.setItem(item, JSON.stringify([checkFor,defaultState]));
   return defaultState;
 }
 
-function showHideAssignments(checkFor, item) {
+function showHideAssignments() {
   const assignmentsTable = document.getElementById("s_m_Content_Content_ExerciseGV");
   const assignmentsTBody = assignmentsTable.getElementsByTagName("tbody");
   const assignmentElements = assignmentsTBody[0].getElementsByTagName("tr");
 
-  let assignmentState = "table-row";
+  const items = JSON.parse(localStorage.getItem("items"))
 
-  if (localStorage.getItem(item) === "true") {
-    assignmentState = "none";
-  } else {
-    assignmentState = "table-row";
+  for (let i = 0; i < assignmentElements.length; i++) {
+    const assignmentTD = assignmentElements[i].getElementsByTagName("td");
+    assignmentElements[i].style.display = "table-row"
   }
 
-  for (let i = 1; i < assignmentElements.length; i++) {
-    const assignmentTD = assignmentElements[i].getElementsByTagName("td");
-    if (assignmentTD[5].innerText === checkFor) {
-      assignmentElements[i].style.display = assignmentState;
-    } else if (assignmentTD[7].innerText === checkFor) {
-      assignmentElements[i].style.display = assignmentState;
+  for (let o = 0; o < items.length; o++) {
+
+    const item = JSON.parse(localStorage.getItem(items[o]))
+    const checkFor = item[0]
+    const state = item[1]
```

Comment:
With the many loops in your code, handling the items outside of the loop, using ``map``, would improve readability.
```js
// LOCATION: Line 186
const localStorageItems = JSON.parse(localStorage.getItem("items"));
const items = localStorageItems.map(item => JSON.parse(localStorage.getItem(item)));
```
That would allow you to do the following: 
```suggestion
    const checkFor = items[o][0];
    const state = items[o][1];
```